packages/flutter/lib/src/gestures/mouse_cursor.dart

```diff
@@ -0,0 +1,332 @@
+    assert(implemented);
+    return implemented;
+  }
+}
+
+/// The base class of a manager that maintains states related to mouse cursor
+/// and provides a simple interface to operate [MouseCursor]s.
+///
+/// Widgets should not use [MouseCursorManager] directly, instead they should
+/// assign [MouseCursor]s to regions, and then [MouseTracker] will handle cursor
+/// changes accordingly.
+///
+/// See also:
+///
+///  * [MouseCursor], which talks more about handling mouse cursors.
+///  * [StandardMouseCursorManager], which implements the platform-specific
+///    code based on the platform that this program is running on.
+///  * [MouseTracker], which uses this class.
+abstract class MouseCursorManager {
+  /// The delegate of the platform that this manager operates.
+  ///
+  /// It is provided to [MouseCursor] to perform platform operations.
+  MouseCursorPlatformDelegate get platformDelegate;
+
+  /// Set the cursor of pointer `device` to `cursor`.
+  ///
+  /// This method handles states or fallbacks.
+  ///
+  /// This method resolves if the operation is successful, or throws errors if
+  /// any occur.
+  Future<void> setDeviceCursor(int device, MouseCursor cursor) async {
+    final MouseCursorActivateDetails details = MouseCursorActivateDetails(
+      device: device,
+      platformDelegate: platformDelegate,
+    );
+    final bool implemented = await cursor.activate(details);
+    if (!implemented) {
+      final bool basicImplemented = await SystemMouseCursors.basic.activate(details);
+      assert(basicImplemented);
+    }
+  }
+}
+
+/// A collection of system [MouseCursor]s.
+///
+/// System cursors are mouse cursors that are included in a platform, available
+/// without external resources. [SystemMouseCursors] is a superset of the system
+/// cursors of every platform that Flutter supports. A cursor that is
+/// unimplemented by a platform will fallback to another cursor or the basic
+/// cursor.
+class SystemMouseCursors {
+  /// A special value that tells Flutter to release the control of cursors.
+  ///
+  /// A layer with this value will absorb the search for mouse cursor
+  /// configuration, but the pointer's cursor will not be changed when it enters
+  /// or is hovering this layer. This value is typically used on a platform view
+  /// or other layers that manages the cursor by itself.
+  static const MouseCursor releaseControl = NoopMouseCursor();
+
+  /// Displays no cursor at the pointer.
+  static const MouseCursor none = _SystemMouseCursor(MouseCursorSystemShape.none, 'none');
+
+  /// The platform-dependent basic cursor. Typically the shape of an arrow.
+  ///
+  /// This cursor is the fallback of unimplemented cursors, and guarantees to
+  /// be implemented by all platforms.
+  static const MouseCursor basic = _EnsuredImplementedSystemMouseCursor(
+    MouseCursorSystemShape.basic, 'basic');
+
+  /// A cursor that indicates links or something that needs to be emphasized
```

Comment:
nit: here and below: The first paragraph should be a one sentence summary. Just put the next sentence into its own paragraph/